File: solutions/python/guidos-gorgeous-lasagna/2/lasagna.py
"""Functions for preparing Guido's gorgeous lasagna."""

EXPECTED_BAKE_TIME = 40
PREPARATION_TIME = 2


def bake_time_remaining(elapsed_bake_time):
    """Return remaining bake time in minutes."""
    return EXPECTED_BAKE_TIME - elapsed_bake_time


def preparation_time_in_minutes(number_of_layers):
    """Return preparation time in minutes for the given number of layers."""
    return number_of_layers * PREPARATION_TIME


def elapsed_time_in_minutes(number_of_layers, elapsed_bake_time):
    """Return total elapsed time (prep + bake) in minutes."""
    return preparation_time_in_minutes(number_of_layers) + elapsed_bake_time

File: solutions/julia/lasagna/1/lasagna.jl
# lasagna.jl

# expected bake time (minutes)
const expected_bake_time = 60

# remaining_time(elapsed_minutes) -> minutes left to bake
remaining_time(elapsed_minutes) = expected_bake_time - elapsed_minutes

# preparation_time(layers) -> minutes needed to prepare layers (2 min per layer)
preparation_time(layers) = layers * 2

# total_working_time(layers, elapsed_minutes) -> preparation time + time already in oven
total_working_time(layers, elapsed_minutes) = preparation_time(layers) + elapsed_minutes
